custom_components/tuya_local/devices/vevor_l2_ev_charger.yaml

```diff
@@ -0,0 +1,223 @@
+name: L2 EV charger
+products:
+  - id: onb9rrxfl9ywgzrg
+    name: VEVOR L2 48A EV Charger
+primary_entity:
+  entity: sensor
+  name: Status
+  class: enum
+  icon: "mdi:ev-station"
+  dps:
+    - id: 3
+      type: string
+      name: sensor
+      mapping:
+        - dps_val: charger_free
+          value: Available
+        - dps_val: charger_insert
+          value: Plugged in
+        - dps_val: charger_free_fault
+          value: Fault (unplugged)
+        - dps_val: charger_wait
+          value: Waiting
+        - dps_val: charger_charging
+          value: Charging
+        - dps_val: charger_pause
+          value: Paused
+        - dps_val: charger_end
+          value: Finished
+        - dps_val: charger_fault
+          value: Fault
+    - id: 23
+      type: string
+      name: system_version
+    - id: 33
+      type: string
+      optional: true
+      name: mode_set
+secondary_entities:
+  - entity: sensor
+    class: energy
+    dps:
+      - id: 1
+        type: integer
+        name: sensor
+        unit: kWh
+        class: total_increasing
+        mapping:
+          - scale: 100
+  - entity: number
+    name: Charge current
+    category: config
+    class: current
+    dps:
+      - id: 4
+        type: integer
+        name: value
+        unit: A
+        range:
+          min: 8
+          max: 48
+  - entity: sensor
+    name: Single phase power
```

Comment:
Ah yeah, I didn't really understand this one (e.g. why there is the power for 9, and then the separate "phase_a" power for 6 as part of the bitmasked value. For this EV charger, it only exists as 2-phase (240v), and both the 6 and 9 values still match identically for me, and match what the app shows (and is the expected value) - I wondered if leaving one out was best, but they both are populated... Maybe this is more useful in chargers that could be single or dual-phase?

I'm not exactly sure how to do what you suggest (very new to Tuya still) - e.g. "add it as an attribute" - would you want me to make these changes or are you able to amend? If it is better if I make them, are there any examples of other devices with similar attribute use I could learn from?

Reply:
I think because your device is not a 3-phase device with independant connection to all 3 phases, but a 240V device connected across 2 phases with a single connection. So it cannot monitor the individual phases.